gaphor/UML/actions/activitypropertypage.py

```diff
@@ -206,47 +207,74 @@ def _on_name_changed(self, entry):
             self.subject.parameter.name = entry.get_text()
 
 
-@PropertyPages.register(ActivityParameterNodeItem)
+@PropertyPages.register(UML.ActivityParameterNode)
 class ActivityParameterNodeTypePropertyPage(TypedElementPropertyPage):
     @property
     def typed_element(self):
-        return self.item.subject.parameter
+        return self.subject.parameter
 
 
 @PropertyPages.register(ActivityParameterNodeItem)
+class ShowActivityParameterNodeTypePropertyPage(ShowTypedElementPropertyPage):
+    @property
+    def typed_element(self):
+        return self.item.subject.parameter
+
+
+@PropertyPages.register(UML.ActivityParameterNode)
 class ActivityParameterNodeDirectionPropertyPage(PropertyPageBase):
     DIRECTION = UML.Parameter.direction.values
     order = 40
 
-    def __init__(self, item):
+    def __init__(self, subject: UML.ActivityParameterNode):
         super().__init__()
-        self.item = item
+        self.subject = subject
 
     def construct(self):
-        if not (self.item.subject and self.item.subject.parameter):
+        if not (self.subject and self.subject.parameter):
             return
 
         builder = new_builder(
             "parameter-direction-editor",
             signals={
                 "parameter-direction-changed": (self._on_parameter_direction_changed,),
-                "show-direction-changed": (self._on_show_direction_changed,),
             },
         )
 
         direction = builder.get_object("parameter-direction")
-        direction.set_selected(
-            self.DIRECTION.index(self.item.subject.parameter.direction)
-        )
-
-        show_direction = builder.get_object("show-direction")
-        show_direction.set_active(self.item.show_direction)
+        direction.set_selected(self.DIRECTION.index(self.subject.parameter.direction))
 
         return builder.get_object("parameter-direction-editor")
 
     @transactional
     def _on_parameter_direction_changed(self, dropdown, _pspec):
-        self.item.subject.parameter.direction = self.DIRECTION[dropdown.get_selected()]
+        self.subject.parameter.direction = self.DIRECTION[dropdown.get_selected()]
+
+
+@PropertyPages.register(ActivityParameterNodeItem)
+class ShowActivityParameterNodeDirectionPropertyPage(PropertyPageBase):
+    order = 40
+
+    def __init__(self, item: ActivityParameterNodeItem):
+        super().__init__()
+        self.item = item
+
+    def construct(self):
+        if not (self.item.subject and self.item.subject.parameter):
+            return
+
+        builder = new_builder(
+            "show-parameter-direction-editor",
+            signals={
+                "show-direction-changed": (self._on_show_direction_changed,),
+            },
+        )
+
+        show_direction = builder.get_object("show-direction")
+        show_direction.set_active(self.item.show_direction)
+
+        print("(self.item.subject and self.item.subject.parameter)")
```

Comment:
One more print statement nit

Reply:
Ah no. I need a linting rule that errors on print statements!